### ISSUE TITLE
Unable to build Ghidra on Windows 10

### ISSUE THREAD

**Issue by wanitaqing** [NationalSecurityAgency/ghidra#2757]
Hi, I'm back again with a similar problem but in a Windows-based environment. 
When I executed `gradle buildGhidra` in my command prompt, I received the following error snippet:

> Task :PDB:win64PDBMake
>
>Executing: msbuild C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.sln /p:Configuration=Release /p:PlatformToolset=v142 /p:WindowsTargetPlatformVersion=
>
>C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB>call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
>**********************************************************************
>** Visual Studio 2019 Developer Command Prompt v16.5.5
>** Copyright (c) 2019 Microsoft Corporation
>**********************************************************************
>[vcvarsall.bat] Environment initialized for: 'x86_x64'
>Microsoft (R) Build Engine version 16.5.1+4616136f8 for .NET Framework
>Copyright (C) Microsoft Corporation. All rights reserved.
>
>Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
>Build started 11/2/2021 12:33:48 pm.
>Project "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Release|x64".
Project "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.sln" (1) is building "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj" (2) on node 1 (default targets).
PrepareForBuild:
  Creating directory "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\..\..\build\tmp\pdb\x64\Release\".
  Creating directory "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\..\..\build\tmp\pdb\x64\Release\pdb.tlog\".
InitializeBuildStatus:
  Creating "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\..\..\build\tmp\pdb\x64\Release\pdb.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\bin\HostX86\x64\CL.exe /c /Zi /nologo /W4 /WX- /diagnostics:column /MP /O2 /D WIN32 /D NDEBUG /D _CONSOLE /D WINVER=0x0601 /Gm- /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\..\..\build\tmp\pdb\x64\Release\\" /Fd"C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\..\..\build\tmp\pdb\x64\Release\vc142.pdb" /Gd /TP /FC /errorReport:queue cpp\err.cpp cpp\find.cpp cpp\iterate.cpp cpp\main.cpp cpp\pdb.cpp cpp\print.cpp cpp\symbol.cpp cpp\xml.cpp
  err.cpp
  find.cpp
C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\find.h(19,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\find.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\err.h(20,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\err.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  iterate.cpp
  main.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\main.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  pdb.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\iterate.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  print.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\pdb.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\print.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  symbol.cpp
  xml.cpp
C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\symbol.h(19,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\symbol.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\xml.h(19,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\xml.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
Done Building Project "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj" (default targets) -- FAILED.
Done Building Project "C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.sln" (default targets) -- FAILED.
>
>Build FAILED.
>
>"C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.sln" (default target) (1) ->
>"C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj" (default target) (2) ->
(ClCompile target) ->
  C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\find.h(19,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\find.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\err.h(20,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\err.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\main.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\iterate.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\pdb.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\yvals.h(12,10): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory (compiling source file cpp\print.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\symbol.h(19,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\symbol.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
  C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\headers\xml.h(19,10): fatal error C1083: Cannot open include file: 'stdio.h': No such file or directory (compiling source file cpp\xml.cpp) [C:\Users\intern\Desktop\Tools\ghidra-debugger\Ghidra\Features\PDB\src\pdb\pdb.vcxproj]
>
>    0 Warning(s)
>    8 Error(s)
>
>Time Elapsed 00:00:02.32
>
> Task :PDB:win64PDBMake FAILED
>
>FAILURE: Build failed with an exception.
>
>* What went wrong:
>Execution failed for task ':PDB:win64PDBMake'.
> Process 'command 'cmd'' finished with non-zero exit value 1
>
>* Try:
>Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
>
>* Get more help at https://help.gradle.org
>
>Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
>Use '--warning-mode all' to show the individual deprecation warnings.
>See https://docs.gradle.org/5.0/userguide/command_line_interface.html#sec:command_line_warnings
>
>BUILD FAILED in 38m 38s
>288 actionable tasks: 288 executed

Here are the details of my Gradle:
![image](https://user-images.githubusercontent.com/47132451/107871470-de843980-6edc-11eb-9569-83842b0675f4.png)

While searching for ways to "install" stdio.h, I found this [answer online](https://stackoverflow.com/a/44754320). However, when I opened the PDB Visual Studio Code Project (I'm using VS 2019), the properties panel is blank. 
![image](https://user-images.githubusercontent.com/47132451/107871478-fcea3500-6edc-11eb-9b61-b2c73d9379bd.png)

I have also followed this [answer](https://stackoverflow.com/a/42778079) but the problem still persist. 

Is it possible that I need to reinstall VS? I would appreciate any help because I was experiencing this problem before building the Ghidra environment on Ubuntu (Hence, the reason I chose to build on Ubuntu) but I then realised that the ghidra debugger (in the ubuntu environment) may not be able to debug windows executable properly (cause may need certain DLLs etc).

**Comment by esaulenka** [comment]
Looks like Gradle cannot detect your Windows SDK.

In my case (Win10 + MSVC2019) everything works good. C++ part compiles with this command line:
```
call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
msbuild C:\Projects\Ghidra\ghidra\Ghidra\Features\PDB\src\pdb\pdb.sln /p:Configuration=Release /p:PlatformToolset=v142 /p:WindowsTargetPlatformVersion=10.0.18362.0
```

Pay attention to `WindowsTargetPlatformVersion` .  I think you need to figure out, what happens in that gradle script:
https://github.com/NationalSecurityAgency/ghidra/blob/ce78b860befe764ed18632962db5c37cf73a64c8/GPL/vsconfig.gradle#L41-L50

**Comment by wanitaqing** [comment]
I actually receive the following output while executing `gradle buildGhidra`
![image](https://user-images.githubusercontent.com/47132451/108040964-ebd62b00-7078-11eb-8766-eaa59d12a422.png)

I've 2 SDKs installed on my system, Windows SDK 8 and 10. Does it affect the default directory ? Because when i tried creating a new VS project (simple printing "Hello World") and built the project, the project managed to build.

**Comment by ryanmkurtz** [comment]
Mine looks like this:
![image](https://user-images.githubusercontent.com/5378554/108061200-f61b0980-7025-11eb-9aa3-bd9fb576e236.png)

I think the best place to start is to reinstall VS2019 to see if that clears up the issue.

**Comment by esaulenka** [comment]
For test I ran `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat amd64`
then ran `set | find "Sdk"` 
I got the following:
```
ExtensionSdkDir=C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs
UniversalCRTSdkDir=C:\Program Files (x86)\Windows Kits\10\
WindowsSdkBinPath=C:\Program Files (x86)\Windows Kits\10\bin\
WindowsSdkDir=C:\Program Files (x86)\Windows Kits\10\
WindowsSdkVerBinPath=C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\
```

**Comment by wanitaqing** [comment]
Hi, 
Sorry for the late reply. I've actually uninstalled and re-installed VS2019 prior to opening this issue. Not sure why the problem isn't resolve after re-installing VS2019. However, I managed to build the debugger compressed file when I built the development environment on a fresh Windows environment, and I just copied that debugger compressed file to my current environment to use debugger. But i received some errors while trying to run the ghidra_dev on the windows environment. I will open a new question for that. :) Thank you!
